libs/auth/src/angular/registration/registration-env-selector/registration-self-hosted-env-config-dialog.component.ts

```
@@ -0,0 +1,153 @@
+import { DialogRef } from "@angular/cdk/dialog";
+import { CommonModule } from "@angular/common";
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from "@angular/core";
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  ValidationErrors,
+  ValidatorFn,
+} from "@angular/forms";
+import { Subject, firstValueFrom } from "rxjs";
+
+import { JslibModule } from "@bitwarden/angular/jslib.module";
+import {
+  EnvironmentService,
+  Region,
+} from "@bitwarden/common/platform/abstractions/environment.service";
+import {
+  AsyncActionsModule,
+  ButtonModule,
+  DialogModule,
+  DialogService,
+  FormFieldModule,
+  LinkModule,
+  TypographyModule,
+} from "@bitwarden/components";
+
+function selfHostedEnvSettingsFormValidator(): ValidatorFn {
+  return (control: AbstractControl): ValidationErrors | null => {
+    const formGroup = control as FormGroup;
+    const baseUrl = formGroup.get("baseUrl")?.value;
+    const webVaultUrl = formGroup.get("webVaultUrl")?.value;
+    const apiUrl = formGroup.get("apiUrl")?.value;
+    const identityUrl = formGroup.get("identityUrl")?.value;
+    const iconsUrl = formGroup.get("iconsUrl")?.value;
+    const notificationsUrl = formGroup.get("notificationsUrl")?.value;
+
+    if (baseUrl || webVaultUrl || apiUrl || identityUrl || iconsUrl || notificationsUrl) {
+      return null; // valid
+    } else {
+      return { atLeastOneUrlIsRequired: true }; // invalid
```

I don't think you need to use a bespoke error summary in your template; instead, you can return an error message here with:

`{ atLeastOneUrlIsRequired: { message: "Foobar error" } }`